netket/graph/graph.py

```diff
@@ -0,0 +1,165 @@
+from abstract_graph import AbstractGraph
+
+import numpy as _np
+import networkx as _nx
+
+class NetworkX(AbstractGraph):
+    """ Wrapper for a networkx graph"""
+    def __init__(self, graph):
+        """
+        Constructs a netket graph from a networkx graph.
+
+        Args: 
+            graph: A networkx graph (might be a networkx.Graph or a networkx.MultiGraph)
+        Examples:
+            A graph of nodes [0,1,2] with edges [(0,1), (0,2), (1,2)]
+            >>> import netket
+            >>> import networkx
+            >>> nx_g = networkx.Graph([(0,1), (0,2), (1,2)])
+            >>> nk_g = netket.graph.NetworkX(nx_g)
+            >>> print(nk_g.n_nodes)
+            3
+        """
+        assert isinstance(graph, _nx.classes.graph.Graph) or isinstance(graph, _nx.classes.multigraph.MultiGraph)
+
+        if isinstance(graph, _nx.classes.graph.Graph):
+            self.graph = _nx.MultiGraph(graph)
+
+        self._automorphisms = None
+
+        super().__init__()
+
+    @property
+    def adjacency_list(self):
+        return [list(self.graph.neighbors(node)) for node in self.graph.nodes]
+
+    @property
+    def is_connected(self):
+        # TODO: how to check if a multigraph is connected?
+        return _nx.is_connected(self.graph)
+
+    def edges(self, color=False):
+        if color is True:
+            return list(self.graph.edges(keys=True))
+        elif color is not False:
+            return [(u, v) for u, v, k in self.graph.edges if k == color]
+        else:
+            return list(self.graph.edges(keys=False))
+
+    def distances(self):
+        return _nx.floyd_warshall_numpy(self.graph)
+
+    @property
+    def is_bipartite(self):
+        return _nx.is_bipartite(self.graph)
+
+    @property
+    def size(self):
+        print("DeprecationWarning: size is deprecated. Use n_nodes from now on.")
```

Comment:
there is a standard way to write deprecations, @femtobit maybe can tell you more about it

Reply:
Is there a need to deprecate it, though? `graph.size` did not exist in NetKet v2 if I recall correctly, so we can just remove it.

Reply:
If I understand correctly, size is being used elsewhere in the code many times. I only added the warning so that existing code runs but points out where the size property is being used. But sure... one can just replace size by n_nodes in the other parts of the code.

Reply:
ok then just remove size, I will then do another PR making sure that size is not used in other places when dealing with the new graph objects